assets/js/index.js

```diff
@@ -0,0 +1,296 @@
+  const forecastWeatherUrl = constructUrl(
+    "https://api.openweathermap.org/data/2.5/onecall",
+    {
+      lat: latitude,
+      lon: longitude,
+      exclude: "minutely,hourly",
+      units: "metric",
+      appid: API_KEY,
+    }
+  );
+
+  // await fetch response
+  const forecastData = await fetchData(forecastWeatherUrl);
+
+  // return data retrieved from api
+  return {
+    name: cityName,
+    weatherData: forecastData,
+  };
+};
+
+const renderWeatherData = async (city) => {
+  try {
+    // fetch weather data
+    const weatherData = await fetchWeatherData(city);
+
+    // empty container
+    weatherContainer.empty();
+
+    // HITTING this
+
+    // render current data
+    renderCurrentWeather(weatherData);
+
+    // NOT hitting this and catching error instead
+
+    // render forecast data
+    renderForecastWeather(weatherData);
+
+    return true;
+  } catch (error) {
+    renderError();
+    return false;
+  }
+};
+
+const renderCurrentWeather = (data) => {
+  // render the current weather data and append to section
+  weatherContainer.append(`<div class="current-weather">
+    <img class="weather-icon" src="http://openweathermap.org/img/w/${
+      data.weatherData.current.weather[0].icon
+    }.png" />
+    <div class="current-weather-info">
+  <h1>${data.name}</h1>
+  <h2>${moment
+    .unix(data.weatherData.current.dt)
+    .format("dddd Do MMMM YYYY")}</h2>
+  <p>Temperature<span>${data.weatherData.current.temp}&deg;C</span></p>
+  <p>Wind<span>${Math.round(
+    data.weatherData.current.wind_speed * 2.23694
+  )}mph</span></p>
+  <p>Humidity<span>${data.weatherData.current.humidity}%</span></p>
+  <p id="indexUV">UV index<span class="${uvIndexColor(
+    data.weatherData.current.uvi
+  )}">${data.weatherData.current.uvi}</span></p>
+    </div>
+    </div>`);
+};
+
+const renderForecastWeather = (data) => {
+  // render the forecast weather data and append each card to section
+  const forecastContainer = $(`<div id="forecast-container"></div>`);
+
+  const createForecastByDate = (each) => {
+    const forecast = `
+    <div class="forecast">
+        <div>
+        <img class="small-weather-icon" src="http://openweathermap.org/img/w/${
+          each.weather[0].icon
+        }.png" />
+        </div>
+        <div class="forecast-weather-info">
+        <h2>${moment.unix(each.dt).format("Do MMM")}</h2>
+        <p><span>${each.temp.day}&deg;C</span></p>
+        <p>Wind<span>${Math.round(each.wind_speed * 2.23694)}mph</span></p>
+        <p>Humidity<span>${each.humidity}%</span></p>
+       
+    </div>`;
+
+    return forecast;
+  };
+
+  const forecastByDate = data.weatherData.daily
+    .slice(1, 6)
+    .map(createForecastByDate);
+
+  forecastContainer.append(forecastByDate);
+
+  weatherContainer.append(forecastContainer);
+};
+
+const renderError = () => {
+  //    remove existing data from container
+  weatherContainer.empty();
+
+  const message = "Oops, that didn't work. Please try again.";
```

Comment:
Not necessary to create a `message` variable here, you're not doing anything with it so may as well just hard code the string into the div:

` weatherContainer.append(`<h2 class="message">Oops, that didn't work. Please try again.</h2>`);`

I've noticed this in another couple of functions, please change those too.